app/resources/api/v2/user_resource.rb

```diff
@@ -15,22 +15,40 @@ module V2
     # or look at the [JSONAPI::Resources](http://jsonapi-resources.com/) package for Sequencescape's implementation
     # of the JSON:API standard.
     class UserResource < BaseResource
-      # Constants...
-
       immutable
 
-      # model_name / model_hint if required
-
-      default_includes :uuid_object
-
+      ###
       # Attributes
+      ###
+
+      # @!attribute [r] uuid
+      #   @return [String] the UUID of the user.
       attribute :uuid, readonly: true
+
+      # @!attribute [r] login
+      #   @return [String] the user's login identifier.
       attribute :login, readonly: true
+
+      # @!attribute [r] first_name
+      #   @return [String] the user's first/given name.
       attribute :first_name, readonly: true
+
+      # @!attribute [r] last_name
+      #   @return [String] the user's last/surname.
       attribute :last_name, readonly: true
 
+      ###
       # Filters
+      ###
+
+      # @!method user_code
+      #   A filter to return only users with the given user code.
+      #   The given user code will be compared with the swipecodes and barcodes for users until matches are found.
       filter :user_code, apply: lambda { |records, value, _options| records.with_user_code(*value) }
+
+      # @!method uuid
+      #   A filter to return only users with the given UUID.
+      filter :uuid, apply: lambda { |records, value, _options| records.with_uuid(*value) }
```

Comment:
Might this Resource want the `creatable_fields` to not include `uuid` too, if it is created by the system on creation?

Reply:
Bearing in mind that I didn't actually create this endpoint, I only added documentation to it and added the UUID filter.

I have been preferring to use `creatable_fields` in more recent resources as it gives nicer error messages than using `readonly: true` on the attribute itself.  You get an error with a description that you cannot include the attribute instead of getting an error 500 from the server when you try to set an attribute that it read only.  I could go through and update all the existing endpoints, but it will mean more messing around with tests so I've avoided doing it for existing resources.

Reply:
Sure, no problem. I thought too it looks much nicer, didn’t know the details but that is definitely better handling. Each resource can be updated as and when they are touched 👍